mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -258,7 +258,70 @@ class Completions(
       case Literal(Constant(null)) :: tl =>
         advancedCompletions(tl, pos, completionPos)
 
+      case (select: Select) :: tail =>
```

Comment:
We could instead of having it here have:
```
lazy val isTypePosition = path match
  ... // all the cases you found
 case _ =>
   false
```

then when in filterInteresting you can add condition:
`!isTypePosition || isTypePosition && sym.isType`
which should filter out all the not needed types.

Then in `completionsWithSuffix` method you can change:
`shouldAddSnippet && sym.is(Flags.Module) && !isJavaDefined`

to
`shouldAddSnippet && sym.is(Flags.Module) && !isJavaDefined && !isTypePosition`

Although probably the change in filterInteresting would filter this out also.

Lastly, findSuffix can have an additional else condition:
 if shouldAddSnippet && methodSymbol.is(Flags.Method) then
  ...
 else if isTypePosition &&  methodSymbol.typeParams.nonEmpty then
   if shouldAddSnippet then Some("[$0]")
   else Some("[]")
 else 
....
```

Reply:
The main issue is finding all the cases for `isTypePosition` which you already did (but that might require some tuning still)

Reply:
Thank you so much for taking the time to add the tips 👍